packages/api/src/command/medical/cq-directory/parse-cq-directory-entry.ts

```diff
@@ -0,0 +1,99 @@
+import { XCA_DQ_STRING, XCA_DR_STRING, XCPD_STRING } from "@metriport/carequality-sdk/common/util";
+import { Address } from "@metriport/carequality-sdk/models/address";
+import { Contained } from "@metriport/carequality-sdk/models/contained";
+import { Organization } from "@metriport/carequality-sdk/models/organization";
+import { normalizeOid } from "@metriport/shared";
+import { CQDirectoryEntryData } from "../../../domain/medical/cq-directory";
+
+export type XCUrls = {
+  urlXCPD: string;
+  urlDQ?: string;
+  urlDR?: string;
+};
+
+export function parseCQDirectoryEntries(orgsInput: Organization[]): CQDirectoryEntryData[] {
+  const orgs = orgsInput.flatMap(org => {
+    const orgOid = org?.identifier?.value?.value;
+    if (!orgOid) return [];
+
+    const url = getUrls(org?.contained);
```

Comment:
Can `org` ever be falsy?

If yes, we could check for it as the first thing and not waste any cycles if that's the case.

If not, we can remove all those question marks throughout the code.

Reply:
I should probably just remove the `.optional()` from the [Bundle schema](https://github.com/metriport/metriport/blob/1265-cq-directory-db/packages/carequality-sdk/src/models/bundle.ts#L10C9-L10C53), no?